src/algebra/module/pointwise_pi.lean

```diff
@@ -0,0 +1,88 @@
+/-
+Copyright (c) 2021 Alex J. Best. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Alex J. Best
+-/
+import algebra.pointwise
+import algebra.module.pi
+
+/-!
+# Pointwise actions on sets in Pi types
+
+This file contains lemmas about pointwise actions on sets in Pi types.
+
+## Tags
+
+set multiplication, set addition, pointwise addition, pointwise multiplication, pi
+
+-/
+
+open_locale pointwise
+open set
+
+@[to_additive]
+lemma smul_univ_pi {K R : Type*} [has_scalar K R] (ι : Type*) {r : K} (t : ι → set R) :
+  r • pi (univ : set ι) t = pi (univ : set ι) (λ (i : ι), r • t i) :=
+begin
+  ext x,
+  simp only [mem_smul_set, mem_univ_pi],
+  split; intro h,
+  { rcases h with ⟨h_w, h_h_left, rfl⟩,
+    simp only [pi.smul_apply],
+    intro i,
+    refine ⟨h_w i, h_h_left i, rfl⟩, },
+  { use (λ i, classical.some (h i)),
+    split,
+    { exact λ i, (classical.some_spec (h i)).left, },
+    { ext i,
+      exact (classical.some_spec (h i)).right, }, },
+end
+
+lemma smul_pi' {K R : Type*} [group_with_zero K] [mul_action K R] {ι : Type*} {r : K}
```

Comment:
You should be able to prove this trivially in terms of `smul_pi` by setting `r` to `units.mk0 r hr`